docs/running-a-node/ledger-management.md

```diff
@@ -0,0 +1,141 @@
+
+## Bootstrapping
+
+When starting a new node the ledger must be downloaded and kept updated in order to participate on the network properly. This is done automatically via bootstrapping - the node downloads and verifies blocks from other nodes across the network. This process can take hours to days to complete depending on network conditions and [hardware specifications](/running-a-node/node-setup/#hardware-recommendations).
+
+### Tuning options
+
+**TODO**: Consider adding notes about tuning here and link to related config entries that need to be updated on configuration page: `bootstrap_connections_max` and `bootstrap_initiator_threads`? Maybe: "Depending on machine and networking resources, the bootstrap performance can be improved by updating X and X configuration values. The additional resource usage these options cause should be considered, especially if left during normal operation (after initial bootstrap is complete)."
+
+## Ledger Fast Sync
+
+!!! tip "Always backup your ledgers file"
+	Whenever you are attempting to change the ledger, it is highly recommended you create backups of the existing `data.ldb` file to ensure you have a rollback point if issues are encountered.
+
+To avoid bootstrapping times, a ledger file (`data.ldb`) can be downloaded off-chain and added to the data file used by the node. The Nano Foundation provides a daily ledger file download in the #ledger channel of our [Discord server](https://chat.nano.org). This is posted by `SergSW` and contains checksums for validation.
+
+Before using this method there are a few considerations to ensure it is done safely:
+
+### Data source
+Make sure you trust the source providing the data to you. If you are unfamiliar with the individual or organization providing the ledger, consider other options for the data or fallback to the default of [bootstrapping](#bootstrapping) from the network.
+
+### Voting weights
+Blocks are confirmed using the voting weight of representatives and these weights are determined by the account balances assigned to those representatives. In addition, the node releases contain a hard-coded set of representative weights captured at the time of release.
+
+If a new ledger is downloaded it is recommended to compare the voting weights it contains against the hard-coded values in the node. To do this, the [--compare_rep_weights](/commands/command-line-interface/#-compare_rep_weights) CLI (_available in V21+ only_) can be run to get an output of the variance.
+
+Although the variance threshold considered to be safe will depend on freshness of the ledger file and network weight variations since the node release date, in general more than a 10% difference should be scrutinized further. If you need support in evaluating join the [Node and Representative Management category](https://forum.nano.org/c/node-and-rep/8) on the [Nano Forums](https://forum.nano.org).
```

Comment:
The variance isn't available in that command currently, however it provides the total mismatched weight and the mean. https://github.com/nanocurrency/nano-node/pull/2719. Not sure how what thresholds should be considered though.

Reply:
Standard deviation is now part of the output here.

Reply:
I am not certain of the right thresholds here either. Considering output such as:

```
ledger weight 115,490,317 Mnano
hardcoded weight 122,578,472 Mnano
mismatched
        total 26,997,517 Mnano
        samples 141
        mean 191,471 Mnano
        sigma 729,315 Mnano
```

can you breakdown what each of these pieces means so we can sort an explanation of how to evaluate?

Reply:
Sure, the difference between total hardcoded and total ledger weight means that more weight is now undelegated (due to pending weight, right?) than in the hardcoded weights. A very large difference here (let's say 20M) and I would be skeptical.

Mismatched is based on individual samples: for every hardcoded rep weight, is it on the provided ledger? Each sample is the absolute difference between those. Total is the accumulation of those differences. As money moves around it is definitely expected to see fluctuations here, but in normal conditions - if weights are well distributed - they shouldn't be large (would indicate one representative receiving a lot of weight).

Sigma is the standard deviation if the sampled mismatched weight is modeled as a gaussian distribution - which it isn't, because it allows negative weight, but good enough for the purpose. [This page](https://en.wikipedia.org/wiki/68%E2%80%9395%E2%80%9399.7_rule) has some useful explanations.

Check the page above. Do you think printing out explicitly any values above 3-sigma (by definition should not be many, 1 or 2 max) is useful here? It could indicate forgery if there is one specific representative that gained a very significant amount of weight. These could then be checked on third party websites.

Moreover, would it be useful to have every mismatch sample printed out, perhaps in the log instead?

Reply:
1. Scan the ledger for integrity using https://docs.nano.org/commands/command-line-interface/#-debug_validate_blocks - if issues are highlighted here inspect carefully and consider looking for alternative sources of a ledger as issues here may be a strong indication of tampering with the ledger.
1. Run `--compare_rep_weights` and review the "outliers" and "new reps" with large amounts, using public explorers to inspect the representatives to make sure the weight changes identified for them match up and that they appear to be valid, participating representatives.

Caveat that this tool isn't giving an absolute answer about issues with a ledger, but can help to identify when additional investigation could be needed. Downloaded ledgers should only ever be used by entities you trust and know as honest actors on the network.

Reply:
I will make adjustments and circulate changes here one latest changes are made per our discussion today.

Reply:
@guilhermelawless Adjustments made based on latest changes to https://github.com/nanocurrency/nano-node/pull/2719, kept it pretty general and thinking we would likely add some details in the CLI command area about the output. Thoughts?

Reply:
Added one comment directly in the changes but otherwise LGTM. Was already thinking of further details in the CLI itself 👍